core/src/main/java/org/frankframework/senders/ParallelSenderExecutor.java

```diff
@@ -65,11 +62,17 @@ public void run() {
 			duration = t2-t1;
 			summary.record(duration);
 		} finally {
-			if (semaphore!=null) {
+			if (semaphore != null) {
 				semaphore.release();
+				log.debug("Released semaphore, available permits: {}", semaphore.availablePermits());
+			}
+			if (countDownLatch != null) {
+				countDownLatch.countDown();
+				log.debug("Counted down latch, remaining: {}", countDownLatch.getCount());
 			}
-			if(guard != null) {
-				guard.releaseResource();
+			if (phaser != null) {
+				phaser.arrive();
+				log.debug("Arrived at phaser, remaining: {}", phaser.getUnarrivedParties());
```

Comment:
Misschien iets minder "technisch", maar iets als `"ShadowSender finished, remaining shadow senders: {}"`

Reply:
Klinkt goed, maar is niet correct. Deze class wordt ook gebruikt vanuit de IteratingPipe en daar is hij geen schaduwzender. Dus laat ik het liever zo.

Reply:
Maar wel een "Sender", dus dan gewoon "Sender" noemen? Ik vind "Arrived at phaser" zo nietszeggend voor wie de log bekijkt.

Reply:
Ja, dat is idd een goed idee en begrijpelijker.